### PR TITLE
allow setting state for postgresql_databases and postgresql_users

### PR DESCRIPTION
**Pull request by mchitten**
this allows configuring the state for databases and users. Default is present as it was previously.

### CODE REVIEW

**Comment by gclough** [comment]
Interesting idea, but you need to fix the syntax errors, and I'd say remove the double quotes around the:

` -- name: PostgreSQL | Make sure the PostgreSQL users are present`

They aren't included elsewhere, so it goes against the style of this role.

File: tests/vars.yml [hunk 1]
```diff
@@ -18,6 +22,11 @@ postgresql_users:
     pass: md51a1dc91c907325c69271ddf0c944bc72
     encrypted: yes
 
+  - name: foo
+    pass: md51a1dc91c907325c69271ddf0c944bc72
+    encrypted: yes
```
Comment (gclough) [inline tests/vars.yml]: Do you need the password, etc to say that a user should be "absent"?

File: tests/docker/group_vars/postgresql.yml [hunk 2]
```diff
@@ -17,6 +21,11 @@ postgresql_users:
     pass: md51a1dc91c907325c69271ddf0c944bc72
     encrypted: yes
 
+  - name: foo
+    pass: md51a1dc91c907325c69271ddf0c944bc72
```
Comment (gclough) [inline tests/docker/group_vars/postgresql.yml]: Do you need a password, etc. to say a user should be "absent"?

File: tasks/users.yml [hunk 3]
```diff
@@ -5,13 +5,13 @@
     name: "{{ postgresql_service_name }}"
     state: started
 
-- name: PostgreSQL | Make sure the PostgreSQL users are present
+- name: "PostgreSQL | Make sure the PostgreSQL users are present"
```
Comment (gclough) [inline tasks/users.yml]: Nowhere else in the role are double-quotes used, so please remove them to maintain the style

File: tasks/databases.yml [hunk 4]
```diff
@@ -14,7 +14,7 @@
     lc_ctype: "{{ item.lc_ctype | default(postgresql_ctype) }}"
     port: "{{postgresql_port}}"
     template: "template0"
-    state: present
+    state: "{{ item.state | default(present)"
```
Comment (gclough) [inline tasks/databases.yml]: This is a syntax error, missing the closing `}}`. right?  Can you please get it to compile and pass the automated tests.

File: README.md [hunk 5]
```diff
@@ -83,6 +84,7 @@ postgresql_users:
   - name: baz
     pass: pass
     encrypted: no       # denotes if the password is already encrypted.
+    state: present      # optional; one of 'present', 'absent'
```
Comment (gclough) [inline README.md]: If this is a string, then we should double-quote it.  In general booleans are unquoted, but all strings should be quoted.

File: README.md [hunk 6]
```diff
@@ -68,6 +68,7 @@ postgresql_databases:
     uuid_ossp: yes      # flag to install the uuid-ossp extension on this database (yes/no)
     citext: yes         # flag to install the citext extension on this database (yes/no)
     encoding: "UTF-8"   # override global {{ postgresql_encoding }} variable per database
+    state: present      # optional; one of 'present', 'absent', 'dump', 'restore'
```
Comment (gclough) [inline README.md]: If this is a string, then we should double-quote it.  In general booleans are unquoted, but all strings should be quoted.

File: tasks/databases.yml (hunk 4 above)
Comment (mchitten) [reply tasks/databases.yml]: thank you! I was looking at this thing for ages trying to figure out where my syntax error was. Figures it was right in front of my face.

File: README.md (hunk 5 above)
Comment (mchitten) [reply README.md]: Sounds good. Does it make sense to make the other string values double quoted in the README too?

**Comment by gclough** [comment]
You're welcome @mchitten.  I'll ask that @jlozadad or @UnderGreen can review it for merge.  I think it makes sense to be able to add as well as remove users, otherwise there is no cleanup possible.

**Comment by mchitten** [comment]
hey @gclough, any update on this? would love to use this for my projects .. thanks!

**Pull request merged**